ibft/instance.go

```diff
@@ -175,33 +175,44 @@ func (i *Instance) ForceDecide(msg *proto.SignedMessage) {
 func (i *Instance) Stop() {
 	// stop can be run just once
 	i.runStopOnce.Do(func() {
-		i.eventQueue.Add(func() {
-			i.Logger.Info("stopping iBFT instance...")
-			i.stopLock.Lock()
-			defer i.stopLock.Unlock()
-
-			i.stopped = true
-			i.roundTimer.Stop()
-			i.SetStage(proto.RoundState_Stopped)
-			i.eventQueue.ClearAndStop()
-
-			// stop stage chan
-			i.stageLock.Lock()
-			defer i.stageLock.Unlock()
-			if i.stageChangedChan != nil {
-				close(i.stageChangedChan)
-				i.stageChangedChan = nil
-			}
-
-			i.Logger.Info("stopped iBFT instance")
-		})
+		if added := i.eventQueue.Add(i.stop); !added {
```

Comment:
@nivBlox I like it less than run once...

Reply:
its not instead. just making sure the msg is added properly